src/py/flwr/cli/run/run.py

```diff
@@ -94,61 +73,62 @@ def run(
 
     typer.secho("Success", fg=typer.colors.GREEN)
 
-    if use_superexec:
-        _start_superexec_run(
-            parse_config_args(config_overrides, separator=","), directory
+    try:
+        federation_name = config["flower"]["federations"]["default"]
+        federation = config["flower"]["federations"][federation_name]
+    except KeyError as err:
+        typer.secho(
+            "❌ The project's `pyproject.toml` needs to declare "
+            "a default federation (with a SuperExec address or an "
+            "`options.num-supernodes` value).",
+            fg=typer.colors.RED,
+            bold=True,
         )
-        return
+        raise typer.Exit(code=1) from err
 
-    server_app_ref = config["flower"]["components"]["serverapp"]
-    client_app_ref = config["flower"]["components"]["clientapp"]
+    if "address" in federation:
 
-    if engine is None:
-        engine = config["flower"]["engine"]["name"]
+        def on_channel_state_change(channel_connectivity: str) -> None:
+            """Log channel connectivity."""
+            log(DEBUG, channel_connectivity)
 
-    if engine == Engine.SIMULATION:
-        num_supernodes = config["flower"]["engine"]["simulation"]["supernode"]["num"]
-        backend_config = config["flower"]["engine"]["simulation"].get(
-            "backend_config", None
+        channel = create_channel(
+            server_address=federation["address"],
+            insecure=True,
+            root_certificates=None,
+            max_message_length=GRPC_MAX_MESSAGE_LENGTH,
+            interceptors=None,
         )
+        channel.subscribe(on_channel_state_change)
+        stub = ExecStub(channel)
+
+        fab_path = build(directory)
+
+        req = StartRunRequest(
+            fab_file=Path(fab_path).read_bytes(),
+            override_config=parse_config_args(config_overrides, separator=","),
+        )
+        res = stub.StartRun(req)
+        typer.secho(f"🎊 Successfully started run {res.run_id}", fg=typer.colors.GREEN)
+
+    else:
+        server_app_ref = config["flower"]["components"]["serverapp"]
+        client_app_ref = config["flower"]["components"]["clientapp"]
+
+        try:
+            num_supernodes = federation["options"]["num-supernodes"]
+        except KeyError as err:
+            typer.secho(
+                "❌ The project's `pyproject.toml` needs to declare "
+                "`flower.federations.localhost.options.num-supernodes` if a "
```

Comment:
Not sure if mixing in f-strings works here, but it'd be nice to provide a somewhat copy & paste-able solution here:

```suggestion
                f"`[flower.federations.{federation_name}]`"
                "`options.num-supernodes = 10` if a "
```